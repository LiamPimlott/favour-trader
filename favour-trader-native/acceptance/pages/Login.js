'use strict';

let I;

module.exports = {

    _init() {
        I = require('../../steps_file.js')();
    },

    login(email, password) {
        I.fillField('~Email-Address', email);
        I.fillField('~Password', password );
        I.tap('~Login');
    },

    goToSignup() {
        I.tap('Sign Up')
    }

}
