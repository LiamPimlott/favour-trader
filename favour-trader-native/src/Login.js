import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput, TouchableOpacity, KeyboardAvoidingView, Image } from 'react-native';
import { StackNavigator, NavigationActions } from 'react-navigation';

export default class Login extends React.Component {
	static navigationOptions = { 
		header: null
	}
	render(){
		const { navigate } = this.props.navigation;
		
		return( 

			<KeyboardAvoidingView behavior="padding" style={styles.container}>
				<Image source={require('../images/test.jpg')} style={styles.backgroundImage}></Image>
				<View style={styles.container}>   
						<Text style={styles.title} >Welcome to Favour Trader</Text>
						<Text style={styles.intro}>Login to start trading!</Text> 
					<View>
						<TextInput
							placeholder="username"
							style={styles.input}
							onSubmitEditing={() => this.passwordInput.focus()}
						/>
						
						<TextInput 
							placeholder="password" 
							style={styles.input}
							secureTextEntry
							ref={(input) => this.passwordInput = input}
						/>
						 
						<TouchableOpacity 
							style={styles.buttonContainer}
							onPress = { () => navigate('Home') }    
						>
							<Text style={styles.buttonText}>LOGIN</Text> 
						</TouchableOpacity> 
						
					
					</View> 
				</View> 
					<TouchableOpacity 
						style={styles.buttonSignUp}
						onPress = { () => navigate('SignUp') }
					>
						<Text style={styles.buttonTextAcc}>No Account?</Text>
						<Text style={styles.buttonText}>SIGN UP</Text>
					</TouchableOpacity>
		  </KeyboardAvoidingView>
		  
		);
	}
}

const styles = StyleSheet.create({
	container: {  
		flex: 1,
		backgroundColor: 'transparent',
		alignItems: 'center',
		justifyContent: 'center',
	},
	backgroundImage: {
		position: 'absolute',
		bottom: 0        
	},
	input: { 
		backgroundColor: '#9978C2',
		padding: 20,
		marginBottom: 5,      
		fontSize: 20
	},
	buttonContainer: { 
		backgroundColor: '#6F4B9B',
		justifyContent: 'center',
		height: '20%',
		width: 400,
		marginTop: 10, 
		marginBottom: 5,
		borderRadius: 20
	},
	buttonSignUp: { 
		backgroundColor: 'transparent', 
		justifyContent: 'center', 
		height: 75,
		width: 200,
		borderRadius: 25 
	},
	buttonTextAcc: {
		textAlign: 'center',
		fontSize: 15
	},	
	buttonText: {
		textAlign: 'center',
		fontSize: 25 
	},
	title: {  
	  fontWeight: 'bold',  
	  fontSize: 45 
	}, 
	intro: { 
	  fontSize: 20 
	}
});