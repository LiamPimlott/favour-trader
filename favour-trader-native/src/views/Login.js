import React, { Component } from 'react';
import { StyleSheet, View, Text, TextInput, TouchableOpacity, KeyboardAvoidingView, Alert, Image } from 'react-native';
import AuthService from "../components/AuthService";
import { Input, Icon, Button, Card, Divider } from 'react-native-elements';

export default class Login extends React.Component {
    constructor(){
        super();
        this.state = {
            failedAttempt: false,
            email: '',
            password: '',
        };
        this.authService = new AuthService();
        this.submit = this.submit.bind(this);
    }

    submit() {
        const { navigate } = this.props.navigation;
        this.authService.login(this.state.email, this.state.password)
            .then(res => {
                if (res.success && res.token) {
                    navigate('Home')
                } else {
                    this.setState({
                        failedAttempt: true,
                    });
                    if(this.state.failedAttempt){
                        this.renderErrorText(res.message);
                    }
                }
            })
            .catch(err => {
                Alert.alert('Exception', err.message);
            });
    }

    renderErrorText(message) {
        Alert.alert('Invalid Login', message);
    }

    render(){
        const { navigate } = this.props.navigation;
        return(
            <KeyboardAvoidingView behavior="padding" style={styles.container}>
                <Card
                    title={'Welcome to Favour â‡‹ Trader'}>
                    <Text style={styles.intro}>Login to start trading!</Text>
                    <View style={styles.input}>
                        <Input
                            placeholder='E-mail Address'
                            leftIcon={
                                <Icon
                                    name='mail-outline' />
                            }
                            autoCapitalize={'none'}
                            onChangeText={(text) => this.setState({email: text})}
                        />
                    </View>
                    <View style={styles.input}>
                        <Input
                            placeholder='Password'
                            leftIcon={
                                <Icon
                                    name='lock-outline' />
                            }
                            autoCapitalize={'none'}
                            secureTextEntry={true}
                            onChangeText={(text) => this.setState({password: text})}
                        />
                    </View>

                    <Button
                        backgroundColor='#03A9F4'
                        buttonStyle={styles.button}
                        title='Login'
                        onPress = { this.submit} />

                    <Divider style={styles.divider}/>
                    <View style={styles.footerView}>
                        <Text style={{textAlign: 'center'}}>No account?</Text>
                        <Button
                            clear={true}
                            titleStyle={{ color: "#03A9F4" }}
                            title='Sign Up'
                            onPress = { () => navigate('Signup') } />
                    </View>
                </Card>
            </KeyboardAvoidingView>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: 'transparent',
        alignItems: 'center',
        justifyContent: 'center',
    },
    input: {
        marginTop: 10,
        marginBottom: 15,
    },
    button: {
        width: 300,
        borderRadius: 0,
        marginLeft: 0,
        marginRight: 0,
        marginBottom: 10
    },
    intro: {
        fontSize: 15,
        textAlign: 'center',
        marginBottom: 10
    },
    divider: {
        width: 300,
        backgroundColor: 'grey',
        marginTop: 15,
    },
    footerView: {
        marginTop: 20,
        flexDirection: 'row',
        justifyContent: 'center',
        alignItems: 'center',
    }
});