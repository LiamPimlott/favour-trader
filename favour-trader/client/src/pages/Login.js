import React, { Component } from 'react';
import { Redirect } from 'react-router-dom';
import {Card,  Form, Icon, Input, Button } from 'antd';
import './Login.css';
import {message} from "antd/lib/index";
const FormItem = Form.Item;

class Login extends Component {
    constructor() {
        super();
        this.state = {
            email: "",
            password: "",
            redirect: false,
            failedAttempt: false,
            iconLoading: false,
        };

        this.renderErrorText = this.renderErrorText.bind(this);
        this.submit = this.submit.bind(this);
    }

    componentWillMount() {
        const { authService } = this.props;
        if (authService.loggedIn()) {
            this.setState({
                redirect: true,
            });
        }
    }

    submit() {
        const {authService} = this.props;
        this.props.form.validateFields((err, values) => {
            if (!err) {
                authService.login(
                    values.emailAddress,
                    values.password,
                    )
                    .then(res => {
                        if (res.success && res.token) {
                            this.setState({
                                redirect: true,
                            });
                        } else {
                            this.setState({
                                failedAttempt: true,
                                iconLoading: true
                            });
                            if(this.state.failedAttempt){
                                this.renderErrorText(res.message);
                            }
                        }
                    })
                    .catch(err => {
                        alert(err);
                    });
            }
        })

    }

    renderErrorText(err) {
        return (
            message.error(err, 2 , ()=> {
                this.setState({
                    iconLoading: false
                });
            })
        );
    }

    render() {
        const { getFieldDecorator } = this.props.form;
        const title = <h5>Favor <Icon type={'swap'}/> Trader</h5>;
        return (
            <div className={'container'}>
                {
                    (this.state.redirect) ? (<Redirect to={'/'}/>) : ('')
                }
                <Card title={title} bordered={true} className={'login-form-card'}>
                    <Form className="login-form">
                        <FormItem>
                            {
                                getFieldDecorator('emailAddress', {
                                    rules: [{ required: true, message: 'Please input your E-mail address!' },
                                            {type: 'email', message: 'The input is not valid E-mail!'}],
                                    validateTrigger: 'onBlur'
                                })(
                                <Input
                                    prefix={<Icon type="mail" style={{ color: 'rgba(0,0,0,.25)' }} />}
                                    placeholder="Email Address"
                                    type="email"
                                    name="email"
                                    id="email" />
                                )
                            }
                        </FormItem>
                        <FormItem>
                            {
                                getFieldDecorator('password', {
                                    rules: [{ required: true, message: 'Please input your password!' }],
                                    validateTrigger: 'onBlur',
                                })(
                                    <Input type="password"
                                           name="password"
                                           id="password"
                                           placeholder=" Password"
                                           prefix={<Icon type="lock" style={{ color: 'rgba(0,0,0,.25)' }} />}/>
                                )
                            }
                        </FormItem>
                        <FormItem>
                            <Button type="primary" htmlType="submit"
                                    onClick={this.submit} loading={this.state.iconLoading}  className="login-form-button">
                                Log in
                            </Button>
                            Or <a href="/create-account">register now!</a>
                        </FormItem>
                    </Form>
                </Card>
            </div>
        );
    }
}

const WrappedLoginForm = Form.create()(Login);
export default WrappedLoginForm;