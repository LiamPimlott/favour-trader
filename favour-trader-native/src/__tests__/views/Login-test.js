import 'react-native';
import React from 'react';
import { mount, shallow, configure } from 'enzyme';
import Login from '../../views/Login.js';
import toJson from 'enzyme-to-json'; //added this line
import renderer from 'react-test-renderer';


const Adapter = require("enzyme-adapter-react-16");
configure({adapter : new Adapter()});

let findById = function(tree, placeholder) {
    if(tree.props && tree.props.placeholder === placeholder) {
        return tree
    }
    if(tree.children && tree.children.length > 0)
    {
        let childs = tree.children
        for(let i = 0; i < childs.length; i++)
        {
            let item = findById(childs[i], placeholder)
            if(typeof(item) !== 'undefined') {
                return item
            }
        }
    }
}

describe('Testing Login component', () => {

  it('renders as expected', () => {
    const navigation = { navigate: jest.fn() };
    const tree = shallow(
      <Login navigation={navigation}/>
    );
      expect(toJson(tree)).toMatchSnapshot(); //edited this line
 });

    it('should render Valid input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<Login navigation={navigation}/>).toJSON()

        expect(findById(tree, 'E-mail Address')).toBeDefined()
        expect(findById(tree, 'Password')).toBeDefined()
    })

    it('should render Empty input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<Login navigation={navigation}/>).toJSON()

        expect(findById(tree, '')).toBeUndefined()
    })

    it('should render Invalid input', () => {
        const navigation = { navigate: jest.fn() };
        let tree = renderer.create(<Login navigation={navigation}/>).toJSON()

        //Possible Typos
        expect(findById(tree, 'E-mali Addres')).toBeUndefined()
        expect(findById(tree, 'Passward')).toBeUndefined()
    })

});
