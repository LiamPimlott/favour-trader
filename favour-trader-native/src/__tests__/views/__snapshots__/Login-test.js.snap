// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login component renders as expected 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  keyboardVerticalOffset={0}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Card
    title="Welcome to Favour â‡‹ Trader"
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "marginBottom": 10,
          "textAlign": "center",
        }
      }
    >
      Login to start trading!
    </Text>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        autoCapitalize="none"
        leftIcon={
          <Icon
            color="black"
            name="mail-outline"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        placeholder="E-mail Address"
      />
    </View>
    <View
      style={
        Object {
          "marginBottom": 15,
          "marginTop": 10,
        }
      }
    >
      <Input
        autoCapitalize="none"
        leftIcon={
          <Icon
            color="black"
            name="lock-outline"
            raised={false}
            reverse={false}
            reverseColor="white"
            size={24}
            underlayColor="white"
          />
        }
        onChangeText={[Function]}
        placeholder="Password"
        secureTextEntry={true}
      />
    </View>
    <Button
      TouchableComponent={[Function]}
      buttonStyle={
        Object {
          "borderRadius": 0,
          "marginBottom": 10,
          "marginLeft": 0,
          "marginRight": 0,
          "width": 300,
        }
      }
      clear={false}
      disabled={false}
      iconRight={false}
      loadingProps={
        Object {
          "color": "white",
          "size": "small",
        }
      }
      onPress={[Function]}
      raised={true}
      title="Login"
    />
    <Divider
      style={
        Object {
          "backgroundColor": "grey",
          "marginTop": 15,
          "width": 300,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 20,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        No account?
      </Text>
      <Button
        TouchableComponent={[Function]}
        buttonStyle={
          Object {
            "borderRadius": 3,
          }
        }
        clear={true}
        disabled={false}
        iconRight={false}
        loadingProps={
          Object {
            "color": "white",
            "size": "small",
          }
        }
        onPress={[Function]}
        title="Sign Up"
        titleStyle={
          Object {
            "color": "#03A9F4",
          }
        }
      />
    </View>
  </Card>
</KeyboardAvoidingView>
`;
